app/Http/Controllers/UserController.php

```diff
@@ -2,50 +2,191 @@
 
 namespace Suyabay\Http\Controllers;
 
+use Validator;
+use Hash;
 use Suyabay\User;
 use Suyabay\Role;
 use Suyabay\Invite;
 use Suyabay\Http\Requests;
 use Illuminate\Http\Request;
-use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Response;
 use Illuminate\Database\QueryException;
 use Suyabay\Http\Controllers\Controller;
+use League\Fractal\Manager;
+use League\Fractal\Resource\Collection;
+use Suyabay\Http\Transformers\UserTransformer;
 
 class UserController extends Controller
 {
     protected $mail;
     protected $response;
+    protected $fractal;
+
+    public function __construct(Manager $fractal)
+    {
+        $this->fractal = $fractal;
+    }
 
     /**
      * This method return all users to the calling API endpoint
      */
-    public function getAllUsers(Response $response)
+    public function getAllUsers(Request $request)
+    {
+        $perPage = 3;
+
+        $page = $request->query('page') ? : 1;
+
+        $totalPage = (int) ($perPage*$page)-$perPage;
```

Comment:
@andela-tolotin Change the $totalPage variable to a better name that explains what is really happening there